### PR TITLE
Python: Series to DataFrame

### PR DESCRIPTION
**Pull request by ax3l**
Add new helpers to create Pandas and cuDF dataframes for a single particle species, over all iterations at once.

- [x] add tests

### CODE REVIEW

File: src/binding/python/openpmd_api/DataFrame.py [hunk 1]
```diff
@@ -74,3 +74,111 @@ def particles_to_dataframe(particle_species, slice=None):
...
+    try:
+        import pandas as pd
+    except ImportError:
+        raise ImportError("pandas NOT found. Install pandas for DataFrame "
+                          "support.")
+
+    df = pd.concat(
+        (
+            series.iterations[i]
+            .particles["electrons"]
```
Comment (franzpoeschel) [inline src/binding/python/openpmd_api/DataFrame.py]:
```suggestion
            .particles[species_name]
```

File: src/binding/python/openpmd_api/DataFrame.py [hunk 2]
```diff
@@ -74,3 +74,111 @@ def particles_to_dataframe(particle_species, slice=None):
...
+        import cudf
+    except ImportError:
+        raise ImportError("cudf NOT found. Install RAPIDS for CUDA DataFrame "
+                          "support.")
+
+    cdf = cudf.concat(
+        (
+            cudf.from_pandas(
+                series.iterations[i]
+                      .particles["electrons"]
```
Comment (franzpoeschel) [inline src/binding/python/openpmd_api/DataFrame.py]:
```suggestion
                      .particles[species_name]
```

**Review by franzpoeschel — CHANGES_REQUESTED** [review]
There are two small hardcoded parameters that you probably meant to replace still.
Looks good otherwise, thank you!

File: src/binding/python/openpmd_api/DataFrame.py (hunk 1 above)
Comment (ax3l) [reply src/binding/python/openpmd_api/DataFrame.py]: Good catch!

**Comment by ax3l** [comment]
Ouch, good catch! Thank you, fixed now.

**Review by franzpoeschel — APPROVED** [review]

**Pull request merged**